src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1252,7 +1252,36 @@ EclipseGrid::EclipseGrid(const Deck& deck, const int * actnum)
         else
             return false;
     }
+    
+    bool EclipseGrid::hasEqualDVDEPTHZ(const Deck& deck) {
+        const std::vector<double>& DXV = deck.get<ParserKeywords::DXV>().back().getSIDoubleData();
+        const std::vector<double>& DYV = deck.get<ParserKeywords::DYV>().back().getSIDoubleData();
+        const std::vector<double>& DZV = deck.get<ParserKeywords::DZV>().back().getSIDoubleData();
+        const std::vector<double>& DEPTHZ = deck.get<ParserKeywords::DEPTHZ>().back().getSIDoubleData();
+        if (EclipseGrid::allEqual(DXV) &&
+            EclipseGrid::allEqual(DYV)&&
+            EclipseGrid::allEqual(DZV)&& 
+            EclipseGrid::allEqual(DEPTHZ))
```

Comment:
This condition is equivalent to requiring that all individual cells in the input grid have the same physical size.  Is that really what you want?  It would mean that, at least initially, ALUGrid would not be usable on models other than those that resemble SPE 1 or SPE 10.  Does ALUGrid refinement not support tensor-product grids?